lib/vector-core/src/config/global_options.rs

```diff
@@ -258,4 +269,66 @@ mod tests {
             Err(vec!["conflicting values for 'timezone' found".into()])
         );
     }
+
+    #[test]
+    fn merges_proxy() {
+        // We use the `.http` settings as a proxy for the other settings, as they are all compared
+        // for equality above.
+        let merge = |a, b| merge("proxy.http", a, b, |result| result.proxy.http);
+
+        assert_eq!(merge(None, None), Ok(None));
+        assert_eq!(merge(Some("test1"), None), Ok(Some("test1".into())));
+        assert_eq!(merge(None, Some("test2")), Ok(Some("test2".into())));
+        assert_eq!(
+            merge(Some("test3"), Some("test4")),
+            Err(vec!["conflicting values for 'proxy.http' found".into()])
+        );
+    }
+
+    #[test]
+    fn merges_acknowledgements() {
+        let merge = |a, b| merge("acknowledgements", a, b, |result| result.acknowledgements);
+
+        assert_eq!(merge(None, None), Ok(None.into()));
+        assert_eq!(merge(Some(false), None), Ok(false.into()));
+        assert_eq!(merge(Some(true), None), Ok(true.into()));
+        assert_eq!(merge(None, Some(false)), Ok(false.into()));
+        assert_eq!(merge(None, Some(true)), Ok(true.into()));
+        assert_eq!(merge(Some(false), Some(false)), Ok(false.into()));
+        assert_eq!(merge(Some(true), Some(true)), Ok(true.into()));
+        assert_eq!(merge(Some(false), Some(true)), Ok(false.into()));
+        assert_eq!(merge(Some(true), Some(false)), Ok(true.into()));
```

Comment:
I'm not sure this is right handling. This implies that the rule for conflicting settings for `acknowledgements` is first-one-wins, which is almost certainly not right. Should these result in a conflict or in one overriding the other?

Reply:
I'd say this should result in a conflict.

Configuring two distinct instances of a global acknowledgements setting, and trying to programmatically make a decision on which one has higher precedence, seems far too risky to me.

Reply:
Thanks, I'll update the handling to check this.